examples/sb3/sb3/env/reward.py

```diff
@@ -0,0 +1,45 @@
+import gym
+import numpy as np
+
+from smarts.core.sensors import Observation
+
+
+class Reward(gym.Wrapper):
+    def __init__(self, env: gym.Env):
+        super().__init__(env)
+
+    def reset(self, **kwargs):
+        return self.env.reset(**kwargs)
+
+    def step(self, action):
+        obs, env_reward, done, info = self.env.step(action)
+
+        wrapped_reward = {
+            agent_id: self._reward(obs[agent_id], agent_reward)
+            for agent_id, agent_reward in env_reward.items()
+        }
+
+        return obs, wrapped_reward, done, info
+
+    def _reward(self, obs: Observation, env_reward: np.float64) -> float:
+        reward = 0
+
+        # Penalty for driving off road
+        if obs.events.off_road:
+            reward -= 5
+            return float(reward)
+
+        # Penalty for colliding
+        if len(obs.events.collisions) > 0:
+            reward -= 10
+            return float(reward)
+
+        # Penalty for stopping
+        if obs.ego_vehicle_state.speed < 1:
+            reward -= 1
+            return float(reward)
```

Comment:
Can the penalty values for different events be configurable i.e. set by the user?

Reply:
This is meant to be an RL example using SB3 library, where we have trained an agent to achieve a reasonable performance using a reward function designed by us. 

If the users want to change the reward function or play around with the example, then they would have to change the code by themselves.